### PR TITLE
 Functionize blockvol create code

### PR DESCRIPTION
**Pull request by humblec**
Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

### CODE REVIEW

**Comment by humblec** [comment]
/area/gluster-block

**Comment by humblec** [comment]
@childsb @rootfs Please review. Thanks.

File: gluster/block/cmd/glusterblock-provisioner/glusterblock-provisioner.go [hunk 1]
```diff
@@ -55,6 +55,7 @@ const (
 	heketiOpmode       = "heketi"
 	glusterBlockOpmode = "gluster-block"
 	volIDAnn           = "gluster.org/volume-id"
+	errIDNotFound      = "Id not found"
```
Comment (rootfs) [inline gluster/block/cmd/glusterblock-provisioner/glusterblock-provisioner.go]: ID not found

File: gluster/block/cmd/glusterblock-provisioner/glusterblock-provisioner.go [hunk 2]
```diff
@@ -319,6 +301,32 @@ func (p *glusterBlockProvisioner) Provision(options controller.VolumeOptions) (*
 	return pv, nil
 }
 
+func (p *glusterBlockProvisioner) storeFieldsFromResponse(blockVolName string, cfg *provisionerConfig, blockVol *glusterBlockVolume, iscsiVol *iscsiSpec) error {
+	//Store fields from response to iscsiSpec struct
+	if cfg.opMode == heketiOpmode && blockVol.heketiBlockVolRes != nil {
```
Comment (rootfs) [inline gluster/block/cmd/glusterblock-provisioner/glusterblock-provisioner.go]: validate cfg != nil and glusterBlockVolume

File: gluster/block/cmd/glusterblock-provisioner/glusterblock-provisioner.go [hunk 3]
```diff
@@ -371,113 +379,132 @@ func (p *glusterBlockProvisioner) createVolume(volSizeInt int, blockVol string,
...
+func (p *glusterBlockProvisioner) glusterBlockExecCreate(blockRes *glusterBlockVolume, config *provisionerConfig, sizeStr string, haCountStr string, blockVol string) error {
+	blockRes.heketiBlockVolRes = nil
 
-				glog.Errorf("failed to unmarshal gluster-block command response: %v", unmarshErr)
-				return nil, fmt.Errorf("failed to unmarshal auth response from gluster-block command output: %v", unmarshErr)
-			}
-			if *execBlockRes == nil {
-				return nil, fmt.Errorf("failed to decode gluster-block response")
-			}
+	cmd := exec.Command(
```
Comment (rootfs) [inline gluster/block/cmd/glusterblock-provisioner/glusterblock-provisioner.go]: validate config blockRes, etc

File: gluster/block/cmd/glusterblock-provisioner/glusterblock-provisioner.go [hunk 4]
```diff
@@ -371,113 +379,132 @@ func (p *glusterBlockProvisioner) createVolume(volSizeInt int, blockVol string,
...
+		config.opMode, "create", config.blockModeArgs["glustervol"]+"/"+blockVol,
```
Comment (rootfs) [inline gluster/block/cmd/glusterblock-provisioner/glusterblock-provisioner.go]: validate config.blockModeArgs["glustervol"] exists

**Comment by rootfs** [comment]
there some validation missing after refactoring into functions.

File: gluster/block/cmd/glusterblock-provisioner/glusterblock-provisioner.go (hunk 1 above)
Comment (humblec) [reply gluster/block/cmd/glusterblock-provisioner/glusterblock-provisioner.go]: Done.

File: gluster/block/cmd/glusterblock-provisioner/glusterblock-provisioner.go (hunk 2 above)
Comment (humblec) [reply gluster/block/cmd/glusterblock-provisioner/glusterblock-provisioner.go]: Done.

File: gluster/block/cmd/glusterblock-provisioner/glusterblock-provisioner.go (hunk 3 above)
Comment (humblec) [reply gluster/block/cmd/glusterblock-provisioner/glusterblock-provisioner.go]: Done.

File: gluster/block/cmd/glusterblock-provisioner/glusterblock-provisioner.go (hunk 4 above)
Comment (humblec) [reply gluster/block/cmd/glusterblock-provisioner/glusterblock-provisioner.go]: Done.

**Comment by humblec** [comment]
@rootfs Thanks for review! I have addressed the comments.

**Comment by humblec** [comment]
@rootfs @childsb PTAL :)

**Pull request merged**